examples/dialogflow-webhook-example/main.py

```diff
@@ -0,0 +1,445 @@
+    flattener = kwargs['flattener']
+    messages = Messages()
+
+    # Get the user_id from the context in the request.
+    user_id = self._get_user_id(flattener)
+
+    # Get the firestore client and extract the user data base on user_id
+    # provided.
+    db = get_firestore_client()
+    client_ref = db.collection(u'users').where(u'user_id', u'==', user_id)
+    users = client_ref.stream()
+    for user in users:
+      # Example:
+      # user_dict: {
+      #   'user_id': 123456,
+      #   'first_name': 'Pedro',
+      #   'last_name': 'Perez',
+      #   'accounts': {
+      #       'checking': {
+      #         'transactions': [
+      #             {'amount': 50, 'type': 'deposit'},
+      #             {'type': 'withdraw', 'amount': '-10'}
+      #           ],
+      #           'balance': 150
+      #        },
+      #        'saving': {
+      #           'transactions': [
+      #               {'amount': 20, 'type': 'deposit'},
+      #               {'type': 'deposit', 'amount': 90}
+      #            ],
+      #            'balance': 110
+      #         }
+      #     }
+      # }
+      user_dict = user.to_dict()
+      logging.info('user_dict: %s', user_dict)
+
+      # Get all the transactions in the checking account.
+      df_checking_transactions = json_normalize(
+          user_dict['accounts']['checking']['transactions'])
+
+      # Get all the transactions in the saving account.
+      df_saving_transactions = json_normalize(
+          user_dict['accounts']['saving']['transactions'])
+
+      # Concatenate all the transactions in one dataframe
+      df_all_transactions = pd.concat(
+          [df_checking_transactions, df_saving_transactions])
+
+      message = 'From webhook: Here are all the transactions that I found:\n'
+      parameters = flattener.get_parameters()
+      transaction_type = parameters['transaction_type'][0]
+
+      # Extract the type of transactions that the user ask: deposit or withdraw.
+      results = df_all_transactions.loc[df_all_transactions.type ==
+                                        transaction_type]
+
+      # Build the message.
+      for row in results.head().itertuples():
+        message = '{0} {1}: {2} '.format(message, row.type, row.amount)
+      return json.dumps({
+          'fulfillmentText': ' '.join([message, 'What else can I do for you?'])
+      })
+    return messages.user_id_not_found(user_id)
+
+  def _get_user_id(self, flattener: Flattener):
+    context_full_name = os.path.join(flattener.get_session(), 'contexts',
+                                     'user_id_action-followup')
+    parameter_full_name = '.'.join(['parameters', 'user_id'])
+    output_context_df = flattener.get_context()
+    user_id = output_context_df.loc[output_context_df.name == context_full_name,
+                                    parameter_full_name].to_list()[0]
+    return user_id
+
+
+class Intents:
+  """Defines intents."""
```

Comment:
Mention that this is a model of dialog flow intents and add link to public docs.